x-pack/test/api_integration/deployment_agnostic/services/alerting_api.ts

```diff
@@ -1096,5 +1106,103 @@ export function AlertingApiProvider({ getService }: DeploymentAgnosticFtrProvide
+        .get('/api/alerting/rules/_find')
+        .query({ filter })
+        .set(roleAuthc.apiKeyHeader)
+        .set(samlAuth.getInternalRequestHeader());
+    },
+
+    async deleteRuleById({ roleAuthc, ruleId }: { roleAuthc: RoleCredentials; ruleId: string }) {
+      return supertestWithoutAuth
+        .delete(`/api/alerting/rule/${ruleId}`)
+        .set(roleAuthc.apiKeyHeader)
+        .set(samlAuth.getInternalRequestHeader());
+    },
+
+    async deleteRules({ roleAuthc, filter }: { roleAuthc: RoleCredentials; filter: string }) {
+      const response = await this.searchRules(roleAuthc, filter);
+      return Promise.all(
+        response.body.data.map((rule: any) => this.deleteRuleById({ roleAuthc, ruleId: rule.id }))
+      );
+    },
+
+    async deleteAllActionConnectors({ roleAuthc }: { roleAuthc: RoleCredentials }): Promise<any> {
+      const res = await supertestWithoutAuth
+        .get(`/api/actions/connectors`)
+        .set(roleAuthc.apiKeyHeader)
+        .set(samlAuth.getInternalRequestHeader());
+
+      const body = res.body as Array<{ id: string; connector_type_id: string; name: string }>;
+      return Promise.all(
+        body.map(({ id }) => {
+          return this.deleteActionConnector({
+            roleAuthc,
+            actionId: id,
+          });
+        })
+      );
+    },
+
+    async deleteActionConnector({
+      roleAuthc,
+      actionId,
+    }: {
+      roleAuthc: RoleCredentials;
+      actionId: string;
+    }) {
+      return supertestWithoutAuth
+        .delete(`/api/actions/connector/${actionId}`)
+        .set(roleAuthc.apiKeyHeader)
+        .set(samlAuth.getInternalRequestHeader());
+    },
+
+    async cleanUpAlerts({
+      roleAuthc,
+      ruleId,
+      consumer,
+      alertIndexName,
+      connectorIndexName,
+    }: {
+      roleAuthc: RoleCredentials;
+      ruleId: string;
+      consumer?: string;
+      alertIndexName?: string;
+      connectorIndexName?: string;
+    }) {
+      return Promise.allSettled([
+        // Delete the rule by ID
+        this.deleteRuleById({ roleAuthc, ruleId }),
+        // Delete all documents in the alert index if specified
+        alertIndexName
+          ? es.deleteByQuery({
+              index: alertIndexName,
+              conflicts: 'proceed',
+              query: { match_all: {} },
+            })
+          : Promise.resolve(),
+        // Delete event logs for the specified consumer if provided
+        consumer
+          ? es.deleteByQuery({
+              index: '.kibana-event-log-*',
+              query: { term: { 'kibana.alert.rule.consumer': consumer } },
+            })
+          : Promise.resolve(),
+        // Delete connector index if provided
+        connectorIndexName
+          ? es.indices.delete({ index: connectorIndexName }).catch((e) => {
+              if (e instanceof errors.ResponseError && e.statusCode === 404) {
+                return;
+              }
+
+              throw e;
```

Comment:
This if-return-throw logic kinda confuses me. 
I guess I was expecting a `Promise.reject()`.
Why throw instead?  Just curious really.

Reply:
This `throw` will result in a `Promise.reject`. What this code-block in ensuring is that it will only reject the promise if the status code is different from 404.

Reply:
Ah thanks Carlos!